io_ogre/ogre/material.py

```diff
@@ -64,23 +67,37 @@ def dot_material(mat, path, **kwargs):
       * touch_textures - bool. Copy the images along to the material files.
     """
     prefix = kwargs.get('prefix', '')
-    generator = OgreMaterialGenerator(mat, prefix=prefix)
+    generator = OgreMaterialGenerator(mat, path, prefix=prefix)
+    # Generate before copying textures to collect images first
+    material_text = generator.generate()
     if kwargs.get('copy_programs', config.get('COPY_SHADER_PROGRAMS')):
-        generator.copy_programs(path)
+        generator.copy_programs()
     if kwargs.get('touch_textures', config.get('TOUCH_TEXTURES')):
-        generator.copy_textures(path)
-    material_text = generator.generate()
+        generator.copy_textures()
     with open(join(path, generator.material_name + ".material"), 'wb') as fd:
         fd.write(bytes(material_text,'utf-8'))
 
     return generator.material_name
 
 class OgreMaterialGenerator(object):
-    def __init__(self, material, prefix=''):
+    # Texture wrapper attribute names
+    TEXTURE_KEYS = [
+        "base_color_texture",
+        "specular_texture",
+        "roughness_texture",
+        "alpha_texture",
+        "normalmap_texture",    # useless ?
```

Comment:
no, quite useful actually. You can tell ogre about it with:
https://ogrecave.github.io/ogre/api/latest/rtss.html#lighting_stage

however, this is out of scope for this PR